caikit/runtime/work_management/abortable_action.py

```diff
@@ -25,6 +27,21 @@
 log = alog.use_channel("ABORT-ACTION")
 
 
+class ActionAborter(abc.ABC):
```

Comment:
Adding this new abstract class was not strictly necessary. I thought I might need an alternate implementation for aborting the internal load thread and found that I didn't. That said, having this as an abstraction will allow a non-grpc-context implementation for HTTP, so I kept it.